docs: scalar funcs update (#4866)
confluentinc/ksql: docs-md/developer-guide/ksqldb-reference/scalar-functions.md

`CONCAT(col1, '_hello')`

Concatenate two or more strings.

EXTRACTJSONFIELD
----------------

`EXTRACTJSONFIELD(message, '$.log.cloud')`

Given a STRING that contains JSON data, extract the value at the specified [JSONPath](https://jsonpath.com/).

For example, given a STRING containing the following JSON:

```json
{
   "log": {
      "cloud": "gcp836Csd",
      "app": "ksProcessor",
      "instance": 4
   }
}
```

> `EXTRACTJSONFIELD(message, '$.log.cloud')` will return the STRING `gcp836Csd`.

Reviewer: ```suggestion
`EXTRACTJSONFIELD(message, '$.log.cloud')` returns the STRING `gcp836Csd`.

```

If the requested JSONPath does not exist, the function returns NULL.

The result of EXTRACTJSONFIELD is always a STRING. Use `CAST` to convert the result to another
type. For example, `CAST(EXTRACTJSONFIELD(message, '$.log.instance') AS INT)` will extract the
instance number from the above JSON object as a INT.

!!! note
    EXTRACTJSONFIELD is useful for extracting data from JSON where either the schema of the JSON
    data is not static, or where the JSON data is embedded in a row encoded using a different

>     format, For example a JSON field within an Avro encoded message.

Reviewer: ```suggestion
    format. For example, a JSON field within an Avro-encoded message.

```

>     Where the whole row is encoded as JSON of a known schema or structure, use the `JSON` format and

Reviewer: ```suggestion
    If the whole row is encoded as JSON with a known schema or structure, use the `JSON` format and

```

    define the structure as the source's columns.  For example, a stream of JSON objects similar to

>     the example above could be defined using something similar to

Reviewer: ```suggestion
    the example above could be defined using a statement similar to this:

```
